mypyc/emitmodule.py

```diff
@@ -104,32 +177,62 @@ def encode_bytes_as_c_string(b: bytes) -> Tuple[str, int]:
     return '"{}"'.format(escaped), len(b)
 
 
-class ModuleGenerator:
+def pointerize(s: str, name: str) -> str:
+    """Given C type and a variable name, return a declaration of a pointer to it."""
+    # This doesn't work in general but does work for all our types...
+    if '(' in s:
+        return s.replace(name, '(*{})'.format(name))
+    else:
+        return s.replace(name, '*{}'.format(name))
+
+
+class GroupGenerator:
     def __init__(self,
                  literals: LiteralsMap,
                  modules: List[Tuple[str, ModuleIR]],
                  source_paths: Dict[str, str],
-                 shared_lib_name: Optional[str],
+                 group_name: Optional[str],
+                 group_map: Dict[str, Optional[str]],
+                 names: NameGenerator,
                  multi_file: bool) -> None:
+        """Generator for C source for a compilation group.
+
+        We do this by just regexping the source, which is a bit simpler than
```

Comment:
Explain what the generated C source includes. This can be understood (based on just this docstring) as being some grepped subset of some other source. Also this would be clearer if the distinction between the "source source" and generated source was more explicit. Maybe a few more sentences would help here?

Reply:
Oh lurr that regexping comment got copy-pasted from elsewhere and is just wrong